lib/pure/times.nim

```diff
@@ -8,223 +8,208 @@
+`Time` type, since it only involves basic arithmetic. Because `Duration`
+is more performant and easier to understand it should generally preferred.
+
+TimeInterval
+----------------------------
+A `TimeInterval` represents an amount of time expressed in calendar
+units, for example "1 year and 2 days". Since some units cannot be
+normalized (the length of a year is different for leap years for example),
+the `TimeInterval` type uses separate fields for every unit. The
+`TimeInterval`'s returned from this module generally don't normalize
+**anything**, so even units that could be normalized (like seconds,
+milliseconds and so on) are left untouched.
+
+Arithmetic with a `TimeInterval` can be very slow, because it requires
+timezone information.
+
+Since it's slower and more complex, the `TimeInterval` type should be
+avoided unless the program explicitly needs the features it offers that
+`Duration` doesn't have.
+
+How long is a day?
+----------------------------
+It should be especially noted that the handling of days differs between
+`TimeInterval` and `Duration`. The `Duration` type always treats a day
+as exactly 86400 seconds. For `TimeInterval`, it's more complex.
+
+As an example, consider the amount of time between these two timestamps, both
+in the same timezone:
+
+  - 2018-03-25T12:00+02:00
+  - 2018-03-26T12:00+01:00
+
+If only the date & time is considered, it appears that exactly one day has
+passed. However, the UTC offsets are different, which means that the
+UTC offset was changed somewhere in between. This happens twice each year for
+timezones that use daylight savings time. Because of this change, the amount
+of time that has passed is actually 25 hours.
+
+The `TimeInterval` type uses calendar units, and will say that exactly one
+day has passed. The `Duration` type on the other hand normalizes everything
+to seconds, and will therefore say that 90000 seconds has passed, which is
+the same as 25 hours.
+
+See also
+========
+* `times/core module <core.html>`_
+* `times/durations module <durations.html>`_
+* `monotimes module <monotimes.html>`_
 ]##
 
-import strutils, math, options
+import std/[strutils, math, options, importutils]
 
 import std/private/since
 include "system/inclrtl"
 
+import ../std/times/core {.all.}
+import ../std/times/durations {.all.}
+export core, durations
+privateAccess(Duration)
+
 when defined(js):
   import jscore
 
-  # This is really bad, but overflow checks are broken badly for
-  # ints on the JS backend. See #6752.
-  {.push overflowChecks: off.}
-  proc `*`(a, b: int64): int64 =
-    system.`*`(a, b)
-  proc `*`(a, b: int): int =
-    system.`*`(a, b)
-  proc `+`(a, b: int64): int64 =
-    system.`+`(a, b)
-  proc `+`(a, b: int): int =
-    system.`+`(a, b)
-  proc `-`(a, b: int64): int64 =
-    system.`-`(a, b)
-  proc `-`(a, b: int): int =
-    system.`-`(a, b)
-  proc inc(a: var int, b: int) =
-    system.inc(a, b)
-  proc inc(a: var int64, b: int) =
-    system.inc(a, b)
-  {.pop.}
-
```

Comment:
I decided to remove this for now, to make things easier. This still needs to be addressed somehow though, perhaps by disabling overflow checks for the JS backend in `system`?

Reply:
this would be addressed by https://github.com/timotheecour/Nim/issues/762 (`int128` which can be implemented on top of jsBigInt on js backend)